proxy/src/cancellation.rs

```diff
@@ -29,14 +32,23 @@ pub struct CancellationHandler<P> {
     /// This field used for the monitoring purposes.
     /// Represents the source of the cancellation request.
     from: CancellationSource,
+    // rate limiter of cancellation requests
+    limiter: Arc<std::sync::Mutex<GlobalRateLimiter>>,
```

Comment:
I was thinking about the limit per IP subnet, but I thought it could be less effective if there is a global brute force from different subnets. Anyway, I believe that moving keyspace to the redis (in another issue PR) and having limits per redis client (as we have it right now) as well will prevent some critical issues, so probably you're right